### PR TITLE
Validate hare output

### PR DESCRIPTION
**Pull request by gavraz**

### CODE REVIEW

File: cmd/node/node.go [hunk 1]
```diff
@@ -366,7 +366,22 @@ func (app *SpacemeshApp) initServices(nodeID types.NodeId, swarm service.Service
 		hOracle = eligibility.New(beacon, atxdb, BLS381.Verify2, vrfSigner, uint16(app.Config.LayersPerEpoch), lg.WithName("hareOracle"))
 	}
 
-	ha := hare.New(app.Config.HARE, swarm, sgn, nodeID, syncer.IsSynced, msh, hOracle, uint16(app.Config.LayersPerEpoch), idStore, atxdb, clock.Subscribe(), lg.WithName("hare"))
+	// a function to validate we know the blocks
+	validationFunc := func(ids []types.BlockID) bool {
```
Comment (almogdepaz) [inline cmd/node/node.go:370]: you dont have do define it here,
 hare knows mesh mesh has access to the GetBLOCK method why not use it ?

Comment (almogdepaz) [reply cmd/node/node.go:370]: just change orphanBlockProvider and add GetBLock to it also rename it to BlockProvider or   something

File: hare/hare.go [hunk 2]
```diff
@@ -115,19 +125,25 @@ func (h *Hare) isTooLate(id InstanceId) bool {
...
 	blocks := make([]types.BlockID, len(set.values))
 	i := 0
 	for _, v := range set.values {
 		blocks[i] = v.BlockID
 		i++
 	}
+
+	// check validity of the collected output
+	if !h.validate(blocks) {
+		h.Panic("Failed to validate the collected output set")
```
Comment (antonlerner) [inline hare/hare.go:138]: should you panic here? why not log and not continue this process?

Comment (gavraz) [reply hare/hare.go:138]: The task is to panic here XD

File: cmd/node/node.go (hunk 1 above)
Comment (gavraz) [reply cmd/node/node.go:370]: Yes, but the Hare doesn't know the mesh as a mesh. We receive the mesh as a block provider...

**Review by almogdepaz — APPROVED** [review]

**Pull request merged**